src/harness/reference_models/geo/utils.py

```diff
@@ -339,3 +343,90 @@ def PolygonsAlmostEqual(poly_ref, poly, tol_perc=10):
   intersection_polys = poly_ref.intersection(poly)
   return ((GeometryArea(union_polys) - GeometryArea(intersection_polys))
           < tol_perc/100. * GeometryArea(poly_ref))
+
+
+def GetClosestCanadianBorderPoint(latitude, longitude, max_dist_km):
```

Comment:
Will this function be different between the US-CA and US-MEX border checks ? If not, suggest to make it a generic name that is not specific for US-CA border.

Reply:
See previous answer.

If we needed to do both, then we would have one function `GetClosestBorderPoint` handling transparently both borders.
However since only the US/CA border is managed by the specification and the test, then it is better to keep the `GetClosestCanadaBorderPoint`  name as it make it very explicit on the calling code that we only do it for one of the border.